From: Tim Gokcen <hexetic@gmail.com>
Date: Sat, 12 Nov 2011 10:34:41 +0100
Subject: tar preserve

Don't use deprecated tar --preserve option
Instead, use --preserve-permissions & --preserve-order

Bug-Debian: http://bugs.debian.org/605048
---
 src/tla/libarch/archive.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/tla/libarch/archive.c b/src/tla/libarch/archive.c
index ac3d580..7e699f7 100644
--- a/src/tla/libarch/archive.c
+++ b/src/tla/libarch/archive.c
@@ -1114,7 +1114,8 @@ invoke_tar_extract (int * pid_ret)
 
       *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = cfg__gnu_tar;
       *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "-m";
-      *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "--preserve";
+      *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "--preserve-permissions";
+      *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "--preserve-order";
       *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "-zxf";
       *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = "-";
       *(t_uchar **)ar_push ((void **)&argv, 0, sizeof (char *)) = 0;
